### DISCUSSION TITLE
milvus 的 VARCHAR 会随着向量一起被加载到内存吗？

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#27994] [Q&A and General discussion]
或者说，milvus 的标量会随着向量一起被加载到内存吗？还是说标量是在硬盘上的，向量 search done 之后根据 milvus_id 去硬盘上找到对应的标量？

我担心的是，如果标量都一起在内存里面，如果是 int 的还好，也就几个字节，但是如果 VARCHAR 也被加载到内存中，那对内存的浪费就很夸张了，尤其是大段文本

还是说，使用者可以自由指定 VARCHAR 是否跟随向量一起被加载到内存中？

**Comment by xiaofan-luan** [comment]
默认是会全部加载啊，2.3.2版本可以支持开启mmap缓存
后面会支持基于字段mmap
